### ISSUE TITLE
Auth0Lock doesn't work with both options and callback function

### ISSUE THREAD

**Issue by yuanzhou** [auth0/lock#288]
According to the documentation: https://auth0.com/docs/libraries/lock

I should be able to use Auth0Lock this way in Javascript:

````
var lock = new Auth0Lock('xxxxxx', '<account>.auth0.com');

// both options and callback
lock.show(options, function (err, profile, token) {});
````

I tried several times and it just doesn't work. I could do 

````
// passing some options
lock.show(options);
````
or 
````
// provide with a callback `fn` to be invoked
// at success or error authentication
lock.show(function (err, profile, token) {});
````

But just can't get it work with both options and callback function.

**Comment by rolodato** [comment]
What do you mean it doesn't work? Can you be more specific?

**Comment by rolodato** [comment]
Also, see here for an example usage of options and callback at the same time: https://auth0.github.io/playground/

**Comment by yuanzhou** [comment]
Not sure if it has anything to do with the SAML connection, that's what I'm using now. When I just pass a callback function:

````
lock.show(
    function (err, profile, token) {
        if (err) {
            console.log("There was an error.", err);
            return;
        }
       console.log(profile); // SAML response user data
       console.log(token); // JWT, it_token
    }
);
````

I could see the profile and token from Chrome console.

But when I passed some options: 

````
lock.show(
       {
            // options
            dict: {
                // More: https://github.com/auth0/lock/blob/master/i18n/en.json
                signin: {
                    title: "Access PICSURE-CCD",
                    footerText: 'Please enter your Pitt or Harvard HMS email. Then you will be redirected to the University login page.'
                }
            },
            container: 'ccd-login-container',
            icon: 'http://www.ccd.pitt.edu/wp-content/uploads/2015/03/logo-sbg-150.png',
            rememberLastLogin: false,
            responseType: 'code' 
        }, 

        function (err, profile, token) {
            if (err) {
                console.log("There was an error.", err);
                return;
            }
           console.log(profile); // SAML response user data
           console.log(token); // JWT, it_token
        }
    );
````

I couldn't see the console messages, and the Auth0Lock kept saying logging in ... and also popped up a new window.

![capture](https://cloud.githubusercontent.com/assets/195873/12301319/4e6f56f8-b9ed-11e5-918f-8b33f9f6591f.PNG)

**Comment by yuanzhou** [comment]
I think I figured this out by commenting: responseType: 'code'

At the very beginning I tried to use this `code` responseType, then send an ajax call to `/oauth/token` according to the documentation at: https://auth0.com/docs/protocols#3-getting-the-access-token
But that kept telling me 500 error. So I thought maybe I can just get the user profile data and token using this callback function. And it worked.
